eclass/kernel-build.eclass

```diff
@@ -225,6 +233,37 @@ kernel-build_src_configure() {
 	emake O="${WORKDIR}"/modprep "${MAKEARGS[@]}" olddefconfig
 	emake O="${WORKDIR}"/modprep "${MAKEARGS[@]}" modules_prepare
 	cp -pR "${WORKDIR}"/modprep "${WORKDIR}"/build || die
+
+	# Now that we have a release file, set KV_FULL
+	local relfile=${WORKDIR}/build/include/config/kernel.release
+	if [[ -z ${KV_FULL} ]]; then
+		KV_FULL=$(<"${relfile}") || die
+	fi
+
+	# Make sure we are about to build the correct kernel
+	if [[ ${PV} != *9999 ]]; then
+		local expected_ver=$(dist-kernel_PV_to_KV "${PV}")
+		local expected_rel=$(<"${relfile}")
+
+		if [[ ${KV_FULL} != ${expected_ver}* ]]; then
+			eerror "Kernel version mismatch!"
+			eerror "Got:      ${KV_FULL}"
```

Comment:
I think it's actually better to match `expected_rel` here. We don't want to bail out on kernel version not matching `KV_FULL` if `KV_FULL` is wrong. And/or reverse the check order.

Reply:
I think doing both checks makes sense here. Since the check for `expected_ver` is done anyway in `kernel-install.eclass`, I think we should fail early, before building the kernel, instead of wasting all the time and energy needed to build a kernel. I will swap the order of the checks though, that makes sense.

Reply:
Reminder to do the swap.